feat: add a script to generate Portuguese NIFs

vinioliveiras opened the pull request:
Adds support for generating random NIFs, NIF is a Fiscal Number in Portugal

Sometimes you need to generate random NIFs numbers to test some features

I guess this could be added as an option to the faker project, thanks!

vinioliveiras committed:
add nif

ST-DDT commented on src/modules/finance/nif.ts:
```diff
@@ -0,0 +1,37 @@
+    return '' + ((mod === 0 || mod === 1) ? 0 : 11 - mod);
+  }
+  function nif(prefix) {
+    var value = pad(Math.floor(Math.random() * 99999999), 8);
```
This does not produce reproducible values.

ST-DDT reviewed (changes requested):
This looks totally different implementation wise then any of our other code.

Also this needs more explanation what it is actually for.

vinioliveiras committed:
feat: add a script to generate Portuguese NIFs

Adds support for generating random NIFs, 

NIF is a Fiscal Number in Portugal

Sometimes you need to generate random NIFs numbers to test some subscribe features in my company system

I guess this could be added as an option to the faker project, thanks!
